pkg/operator/ceph/controller/cluster_info.go

```diff
@@ -166,10 +172,47 @@ func CreateOrLoadClusterInfo(clusterdContext *clusterd.Context, context context.
 		clusterInfo.CephCred.Username = string(secret.Data["userID"])
 		clusterInfo.CephCred.Secret = string(secret.Data["userKey"])
 	}
+	if cephUsername, ok := secrets.Data[CephUsernameKey]; ok && string(cephUsername) == "client.admin" {
```

Comment:
To preserve the backward compatibility of this change, here are some more details from our discussion:
1. Use new key names in the secret for creds like this:
```go
CephUsernameKey    = "ceph-operator-username"
CephUserSecretKey  = "ceph-operator-secret"
CephLegacyUsernameKey  = "ceph-username"
CephLegacySecretKey         = "ceph-secret"
```
2. For the external cluster scenario, update the secret to copy the keys from `ceph-username` and `ceph-secret` to the new keys so that the external scripts do not need to change.
3. Update the toolbox to use these new key names
4. During an upgrade, confirm that the original values for `ceph-username` and `ceph-secret` are preserved. Then any spec such as the previous version of the toolbox will still work with the old values.